File: app/src/main/java/xlagunas/cat/andrtc/di/modules/UserModule.java
package xlagunas.cat.andrtc.di.modules;


import javax.inject.Named;
import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;
import xlagunas.cat.andrtc.di.PerActivity;
import xlagunas.cat.data.mapper.UserEntityMapper;
import xlagunas.cat.domain.User;
import xlagunas.cat.domain.executor.PostExecutionThread;
import xlagunas.cat.domain.interactor.LoginUseCase;
import xlagunas.cat.domain.interactor.UseCase;
import xlagunas.cat.domain.repository.UserRepository;


/**
 * Created by xlagunas on 2/03/16.
 */

@Module
public class UserModule {

    private User user;

    public UserModule() {}

    public UserModule(User user) {
        this.user = user;
    }

//    @Provides
//    @PerActivity @Named("userList")
//    UseCase provideGetUserListUseCase(
//            GetUserList getUserList) {
//        return getUserList;
//    }

    @Provides
    @PerActivity
    @Named("login")
    UseCase provideGetUserDetailsUseCase(
            UserRepository userRepository, PostExecutionThread postExecutionThread) {
        return new LoginUseCase(postExecutionThread, userRepository);
    }

}

File: user/src/main/java/cat/xlagunas/user/di/UserModule.kt
package cat.xlagunas.user.di

import cat.xlagunas.user.data.AuthenticationApi
import cat.xlagunas.user.data.AuthenticationRepositoryImpl
import cat.xlagunas.user.domain.AuthenticationRepository
import dagger.Module
import dagger.Provides
import retrofit2.Retrofit

@Module
class UserModule {

    @Provides
    fun provideRegisterApi(retrofit: Retrofit): AuthenticationApi {
        return retrofit.create(AuthenticationApi::class.java)
    }

    @Provides
    fun provideRegisterRepository(authenticationRepository: AuthenticationRepositoryImpl): AuthenticationRepository {
        return authenticationRepository
    }
}